ddtrace/tracer/sampler.go

```diff
@@ -24,6 +24,36 @@ import (
 	"golang.org/x/time/rate"
 )
 
+// samplerName specifies the name of a sampler which was
+// responsible for a certain sampling decision.
+type samplerName int8
+
+const (
+	// samplerUpstream specifies that the sampling decision was made in an upstream service
+	// and no sampling needs to be done.
+	samplerUpstream samplerName = math.MinInt8
+	// samplerUnknown specifies that the span was sampled
+	// but, the tracer was unable to identify the sampler.
+	samplerUnknown samplerName = -1
+	// samplerDefault specifies that the span was sampled without any sampler.
+	samplerDefault samplerName = 0
+	// samplerAgentRate specifies that the span was sampled
+	// with a rate calculated by the trace agent.
+	samplerAgentRate samplerName = 1
+	// samplerRemoteRate specifies that the span was sampled
+	// with a dynamically calculated remote rate.
+	samplerRemoteRate samplerName = 2
+	// samplerRuleRate specifies that the span was sampled by the RuleSampler.
+	samplerRuleRate samplerName = 3
+	// samplerManual specifies that the span was sampled manually by user.
+	samplerManual samplerName = 4
+	// samplerAppSec specifies that the span was sampled by AppSec.
+	samplerAppSec samplerName = 5
```

Comment:
@knusbaum, @Barbayar: given this value is used by AppSec but is currently hard-coded there, WDYT of exporting it `/internal/something` so that `/contrib/**/appsec.go` can use the same const value?
I fear this value changes at some point without changing them in appsec, and conversely.

Reply:
👍 I suggested the same thing. If others agree, let's go ahead and move all of these.